File: 792-BinarySearch/792-BinarySearch.py
# Last updated: 8/16/2026, 9:49:17 PM
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        while l < r:
            mid = l + (r-l)//2 
            # '//' is the integer division, / will result to float type
            if nums[mid] == target: return mid
            elif nums[mid] < target: l = mid + 1
            else: r = mid - 1
        
        if nums[l] == target: return l
        
        return -1

File: 792-BinarySearch/792-BinarySearch.java
// Last updated: 8/16/2026, 9:49:22 PM
class Solution {
    public int search(int[] nums, int target) {
        int left = 0, right = nums.length - 1;
        while(left < right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        if(nums[left] == target) return left;
        return -1;
    }
}